article: tutoriel-01: hello world !

Supercip971 opened the pull request.

Supercip971 committed:
article: tutoriel-01: hello world !

Supercip971 committed:
demande à ce que le lecteur utilise un cross compilateur
+ "nous allons essayer de faire" -> "nous allons faire"

sleepy-monax commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,339 @@
+
+> résultat à la fin de ce tutoriel
+
+Dans cette partie nous allons essayer de faire un "hello world !" dans le vga. Cependant vous avez 2 options pour le compilateur, soit vous faites un [cross compilateur](/cross-compilation/creer-un-cross-compiler.md) soit vous utilisez le compilateur fournis par votre distribution linux (gcc), ce n'est pas très recommandé mais vous pouvez toujours repousser l'utilisation du cross compilateur à pl ...
```
Tu vas pas essayer, tu vas le faire

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,339 @@
+    kernel_phys_offset = 0xffffffff80100000;
+    . = kernel_phys_offset;
+    
+    .stivalehdr ALIGN(4K):
```
pourquoi stivale 1 ?

Supercip971 replied:
parce que c'est plus simple pour un hello world

d0p1s4m4 replied:
Mais stivale1 est déprécié et il me semble qu'il était question de ne plus le supporter dans stivale

Supercip971 replied:
non stivale1 n'est pas déprécié (il me semble) limine supporte toujours ce protocole

Supercip971 replied:
dans le dernier commit j'explique stivale2 à la place de stivale1

sleepy-monax commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,339 @@
+- `-no-pie`: Ne produit pas un executable avec une position indépendante.
+- `-masm=intel`: Utilise l'asm intel pour la génération de code.
+
+```makefile
+GCC_C_FLAGS :=              \
+	-Isrc                   \
+	-std=c11                \
+	-ffreestanding          \
+	-fno-stack-protector    \
+	-fno-pic                \
+    -no-pie                 \
+    -O1                     \
+    -m64                    \
+    -g                      \
+    -masm=intel             \
+    -mno-red-zone           \
+    -mno-sse                \
+    -mno-avx                
+```
````
Pas de cross compiler? O.O

Supercip971 replied:
j'ai expliqué qu'ici on pouvait soit faire un cross compiler avec la page de doc que j'avais écrit soit on prend le compilateur de base mais que c'était déconseillé

sleepy-monax replied:
IMO on ne devrais meme pas montrer mauvaise manière

Supercip971 replied:
oki

Supercip971 replied:
fixé

keyboard-slayer commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,339 @@
+
+> résultat à la fin de ce tutoriel
+
+Dans cette partie nous allons essayer de faire un "hello world !" dans le vga. Cependant vous avez 2 options pour le compilateur, soit vous faites un [cross compilateur](/cross-compilation/creer-un-cross-compiler.md) soit vous utilisez le compilateur fournis par votre distribution linux (gcc), ce n'est pas très recommandé mais vous pouvez toujours repousser l'utilisation du cross compilateur à pl ...
```
Ou clang :eyes:

Supercip971 replied:
ta review une ancienne version :eyes:

keyboard-slayer replied:
ah :eyes:

keyboard-slayer commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,339 @@
+- [Limine](https://github.com/limine-bootloader/limine) comme bootloader
+- [Echfs](https://github.com/echfs/echfs) comme système de fichier
+
+Nous utiliserons echfs comme système de fichier car c'est assez simple de l'utiliser quand on débute, car normalement sans echfs on doit créer un disque, le partitionner, le monter, installer un système de fichier, mettre nos fichier... Avec echfs c'est plus simple (grâce à l'outil `echfs-utils`). 
```
Trop de répétitions avec "Nous utiliserons"

Supercip971 committed:
correction: make: utilisation de .SUFFIXE
+ C_FLAGS -> CFLAGS

sleepy-monax commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -44,7 +46,7 @@ Puis ensuite vous pouvez compiler les fichiers C mais avant nous devons changer
 
 ```makefile
-GCC_C_FLAGS :=              \
+C_FLAGS :=                  \
````
Le nom standar c'est CFLAGS

sleepy-monax replied:
Et tu devrais ajouter `.SUFFIXES:` au debut du makefile

Supercip971 replied:
fixé ;)

Supercip971 committed:
article: tutoriel 01, utilisation du protocole stivale2 à la place du protocole stivale1
+ corrections

Supercip971 committed:
Corrections
(correction proposées par d0p1):
- C_FILES -> SRCS
- O_FILES -> OBJS
- utilisation de wildcard
- make system -> système de build
- stivale [2] -> stivale 2
- fix d'un espace dans un titre qui n'était pas un espace (wtf)

Supercip971 committed:
ajout de sources

Supercip971 committed:
Corrections
venant de @Procospecteur (merchi beaucoup <3)

Supercip971 committed:
correction: sources -> référence + ajout d'une source
- ajout de l'article wikipedia sur la stack comme référence car nous en parlons dans l'article

Supercip971 committed:
correction: utilisation de vous à la place de nous

Supercip971 committed:
correction: markdown-lint
- espaces entre le code
- espaces entre les listes
- espaces consécutif
-  fix des niveaux des titres
- espaces entre le texte et les deux points
- fix d'un 'nous'

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+# Le Fichier Makefile
+
+> Note: vous pouvez utiliser d'autres make system, il faut juste suivre les même commandes et arguments pour gcc/ld.
```
"d'autres systèmes de build" plutôt que "make system" non ?

Supercip971 replied:
fixé ;) merchi

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+> Ici nous prenons en compte que vous utiliser le dossier "src" pour mettre le code de votre kernel.
+
+```makefile
+C_FILES := $(shell find ./src/ -type f -name '*.c')
+O_FILES := $(C_FILES:.c=.o)
````
Par convention on utilise en général "SRCS" et "OBJS" 
Après ne serait il pas mieux d'utiliser un [wilcard](https://www.gnu.org/software/make/manual/html_node/Wildcard-Function.html) plutôt qu'un appel à find ?

Supercip971 replied:
parfois on m’a dit find > wildcard et parfois l'inverse

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Puis ensuite vous pouvez compiler les fichiers C mais avant nous devons changer certains flags du compilateurs:
+
+- `-ffreestanding`: Active l'environnement freestanding, cela signifie que le compilateur désactive les librairies standarts du C (faites pour linux). Il signifie aussi que les programmes ne commencent pas forcément à `main`. 
```
standarts -> standards
linux -> GNU/Linux

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Vous commencerez par créer un fichier kernel.c dans le dossier src (le nom du fichier n'est pas obligé d'être kernel.c).
+
+Mais avant nous devons rajouter le header du bootloader, qui permet de donner des informations/configurer le bootloader quand il charge le kernel, ici nous utilisons le protocol stivale [2], nous recommandons d'utiliser [le code/header fournis par stivale2](https://github.com/stivale/stivale/blob/master/stivale2.h) qui facilite la création du header.
```
Pourquoi le 2 entre [] ?

Procospecteur replied:
Correction proposée :
"[...] le protocole stivale [...]"

d0p1s4m4 commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Maintenant il est conseillé de compiler et de tester le kernel, avant de continuer. Faites un `make run` il faut qu'il n'y ait pas d'erreur (que cela soit du bootloader ou de qemu).
+
+## Lire Le Bootloader_data
```
Le titre n'est pas rendu correctement dans la preview (aucune idée de pourquoi) cf image
![image](https://user-images.githubusercontent.com/17814263/116980368-70bf0100-acc6-11eb-9503-4f92a583675c.png)

d0p1s4m4 reviewed (changes requested):
A part les point mentionné le reste me semble bon.

sleepy-monax commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+<img src="/x86_64/assets/tutoriel-hello-world-result.png">
+
+# Conclusion
+Cette partie du tutoriel est terminée ! vous avez maintenant un kernel qui boot, cependant dans le prochain tutoriel nous implémenterons un driver COM, qui donnera la possibilité d'écrire des informations dans la console, ce qui est très pratique pour debugger.
```
Pas de sources ?

d0p1s4m4 replied:
Je pense pas que l'ajout des sources complètes soit pertinente et ça permet d'éviter le Ctrl + C / Ctrl + V qu'on à trop tendance à voir avec les tutoriels

sleepy-monax replied:
je veux dire sources bibliographiques

d0p1s4m4 replied:
my bad, effectivement

Supercip971 replied:
Techniquement il n’y a pas de source, c''est surtout des coups d'œils de code, après je pourrais rajouter les specs de limine /stivale et echfs-utils comme sources

sleepy-monax replied:
Si il y a des sources a moins que tu soit l'inventeur du port com, gcc, binutils, make, etc

Supercip971 replied:
pour un hello world il n'y a pas "grand chose", c'est surtout des liens pour aller plus loin dans les informations que des sources

Supercip971 replied:
j'ai rajouté les sources, c'est fixé, merchi

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Pour commencer vous devez mettre un place un [cross compilateur](/cross-compilation/creer-un-cross-compiler.md) dans votre projet.
+
+Vous utiliserez echfs comme système de fichier car c'est assez simple de l'utiliser quand on débute, car normalement sans echfs on doit créer un disque, le partitionner, le monter, installer un système de fichier, mettre nos fichier... Avec echfs c'est plus simple (grâce à l'outil `echfs-utils`). 
```
Correction proposée :
"Vous utiliserez echfs comme système de fichiers car il est assez simple d'utilisation pour les débutants. Normalement sans echfs, il faut créer un disque, le partitionner, le monter, installer un système de fichiers, ajouter nos fichiers, ... En utilisant echfs et l'outil `echfs-utils`, c'est bien plus simple."

sleepy-monax replied:
En vrais c'est chelou d'utiliser le vous dans un tuto, souvent on fait du nous

Procospecteur replied:
C'est à Supercyp de voir, j'ai uniquement corrigé les tournures de phrases et les quelques rares fautes de grammaire. Mais en soi, oui je suis d'accord avec toi

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Pour commencer nous devons obtenir tout les fichier '.c' avec find et obtenir le fichier objet '.o' équivalent à ce fichier c.
+
+> Ici nous prenons en compte que vous utiliser le dossier "src" pour mettre le code de votre kernel.
```
Correction proposée :
"[...] que vous utilisez [...]"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+O_FILES := $(C_FILES:.c=.o)
+```
+
+Puis ensuite vous pouvez compiler les fichiers C mais avant nous devons changer certains flags du compilateurs:
````
Correction proposée :
"Ensuite, juste avant de compiler les fichiers C, il faut changer certains flags du compilateur :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Maintenant vous pouvez rajouter une target a votre makefile pour compiler vos fichier C en objet:
+
+> Ici nous prenons en comptes que vous avez mit CC soit avec le path de votre cross compilateur.
```
Correction proposée :
"Ici, nous prenons en compte que vous avez mis CC, soit avec le path de votre cross-compilateur"

Remarque : c'est vrai que c'est assez bizarre de mixer le nous et le vous dans un tutoriel. Qui est le "nous" au juste ?

Supercip971 replied:
je crois que je vais remplacer les "nous" par des "vous"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+## Linking
+
+Après avoir compilé tout les fichier C en fichier objet, nous devons les liers pour créer le fichier du kernel.
```
Correction proposée :
"[...] les lier [...]"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Après avoir compilé tout les fichier C en fichier objet, nous devons les liers pour créer le fichier du kernel.
+
+Nous utiliserons `ld` (soit celui fournit par le binutils de votre cross compilateur soit celui fournit par votre distribution)
```
Correction proposée :
"[...] soit celui fourni par votre distribution)."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Nous utiliserons `ld` (soit celui fournit par le binutils de votre cross compilateur soit celui fournit par votre distribution)
+
+Avant il nous faut un fichier de linking, qui définit la position de certaines parties du code. Nous le mettrons dans le chemins `src/link.ld`.
```
Correction proposée :
"[...] dans le chemin [...]"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+ENTRY(kernel_start)
+```
+
+Ensuite il faut définir la position des sections du code (que cela soit les données (data/rodata/bss) ou le code (text)), soit la position `0xffffffff80100000` car c'est un kernel higher half, donc il est placé dans la moitié haute de la mémoire: `0xffffffff80000000` mais ici nous rajoutons un décalage de 1M (`0x100000`) (pour éviter de toucher l'adresse 0 en physique). 
````
Correction proposée :
"Il faut ensuite définir la position des sections du code (pour les données (data/rodata/bss) et le code (text)), soit la position `0xffffffff80100000`. Étant donné que c'est un kernel "higher-half", il est donc placé dans la moitié haute de la mémoire : `0xffffffff80000000`. Ici, nous rajoutons un décalage de 1M (`0x100000`) pour éviter de toucher l'adresse 0 en physique."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Ensuite il faut définir la position des sections du code (que cela soit les données (data/rodata/bss) ou le code (text)), soit la position `0xffffffff80100000` car c'est un kernel higher half, donc il est placé dans la moitié haute de la mémoire: `0xffffffff80000000` mais ici nous rajoutons un décalage de 1M (`0x100000`) (pour éviter de toucher l'adresse 0 en physique). 
+
+Nous devons aussi positioner le header pour le bootloader (ici dans la section `stivale2hdr`), il permet de donner des informations importantes quand le bootloader lit le kernel. Le bootloader demande à ce que cette section soit la première dans le code.
```
Correction proposée :
"[...] positionner le header [...]"
"[...] Le bootloader demande à cette section d'être la première dans le code."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Nous devons aussi positioner le header pour le bootloader (ici dans la section `stivale2hdr`), il permet de donner des informations importantes quand le bootloader lit le kernel. Le bootloader demande à ce que cette section soit la première dans le code.
+
+Pour finir nous avons: 
```
Correction proposée :
"Pour finir, nous avons :"
(espace obligatoire entre le double point et le texte)

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+}
+```
+
+Comme pour la compilation des fichiers C, nous devons passer des arguments spécifiques:
````
Correction proposée :
"[...] des arguments spécifiques :"
(espace obligatoire entre le double point et le texte)

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+
+```
+
+Maintenant nous pouvons lier les fichiers objets en un `kernel.elf`: (en utilisant une nouvelle target dans le fichier Makefile):
````
Correction proposée :
"En utilisant une nouvelle target dans le fichier Makefile, nous pouvons désormais lier les fichiers objets en un `kernel.elf` :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+```
+
+## Création Du Fichier De Configuration Du Bootloader
+Avant de continuer, nous devons créer un fichier `limine.cfg`, c'est un fichier lu par le bootloader, il paramètre certaines options et pointe où se trouve le kernel dans le disque:
````
Correction proposée :
"Avant de continuer, nous devons créer un fichier `limine.cfg`. C'est un fichier lu par le bootloader qui paramètre certaines options et pointe vers où se trouve le kernel dans le disque :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+KERNEL_PATH=boot:///kernel.elf
+```
+
+Ici nous voulons définir l'entrée `mykernel` qui a le protocol `stivale2` et qui a comme fichier elf pour le kernel: `/kernel.elf` dans la partition de `boot`.
````
Correction proposée :
"[...] le protocole [...]"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+## L'Execution
+
+Ici une fois avoir créer le disque nous allons faire une cible: `run`, elle servira plus tard quand vous pourrez enfin tester votre kernel ;) .
```
Correction proposée :
"Une fois le disque créé, nous allons faire une cible : `run`. Elle servira plus tard quand vous pourrez enfin tester votre kernel. ;)"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+# Le Code
+
+Maintenant après avoir tout configuré avec le makefile, nous pouvons commencer à coder ! 
```
Correction proposée :
"Après avoir tout configuré avec le makefile [...]"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+Mais avant nous devons rajouter le header du bootloader, qui permet de donner des informations/configurer le bootloader quand il charge le kernel, ici nous utilisons le protocol stivale [2], nous recommandons d'utiliser [le code/header fournis par stivale2](https://github.com/stivale/stivale/blob/master/stivale2.h) qui facilite la création du header.
+
+Nous créons une variable dans le `kernel.c` du type stivale2_header, nous demandons au linker de la positioner dans la section "`.stivale2hdr`" et de forcer le fait qu'elle soit utilisée (pour éviter à ce que le compilateur vire l'entrée automatiquement).
```
Correction proposée :
"[...] (pour éviter que le compilateur vire l'entrée automatiquement)."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+struct stivale2_header header = {/* entrées */};
+```
+
+Puis nous replissons toutes les entrées.
````
Correction proposée :
"Puis nous remplissons toutes les entrées." ou
"Puis nous définissons toutes les entrées."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+```
+
+Puis nous replissons toutes les entrées.
+Il faut avant créer une variable pour définir la [stack](https://fr.wikipedia.org/wiki/Pile_(informatique)) du kernel. Nous utilisons une stack de taille 32768 (32K) soit:
````
Correction proposée :
"Il faut commencer par créer une variable [...] soit :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+struct stivale2_header header = {.stack = (uintptr_t)kernel_stack + (STACK_SIZE) }// la stack tend vers le bas, donc nous voulons donner le dessus de cette stack
+```
+
+Le header doit spécifier le point d'entrée du kernel par la variable `entry_point`, il faut le mettre à 0 pour demander au bootloader d'utiliser le point d'entrée spécifié par le fichier elf.
````
Question :
Est-ce-que traduire "header" en "en-tête" (ou "entête") est souhaitable ?

Supercip971 replied:
on ne traduit pas vraiment les mots qui sont utilisés comme ça dans le code, par exemple entry_point quand on y fait référence on dit toujours entry point, après peut être la traduction d'entête est meilleure ?

Procospecteur replied:
J'ai déjà vu des tutoriels dire les deux en fait, c'est pour ça que je demandais. Pour ma part, je pense que laisser header c'est bien

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+};
+```
+
+Maintenant il faut mettre en place des tags pour le bootloader, les tags sont une liste liée, càd que chaque entrée doit indiqué ou est la prochaine entrée:
````
Correction proposée :
"[...] c'est à dire que chaque entrée doit indiquer où est la prochaine entrée :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,463 @@
+
+<img src="/x86_64/assets/tutoriel-hello-world-stivale2-linked-list.svg" style="margin:5rem;padding:1rem;width:64rem;background-color:white;">
+
+Il y a plusieurs valeurs valide pour le `identifier` qui identifie l'entrée, et nous pouvons avoir plusieurs tags mais ici nous allons en utiliser qu'un seul, celui pour définir le framebuffer.
```
Correction proposée :
"Il y a plusieurs valeurs valides pour le `identifier` qui identifie l'entrée et nous pouvons avoir plusieurs tags. Ici nous n'allons en utiliser qu'un seul : celui pour définir le framebuffer."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+    },
+}; 
+```
+Ici la variable `.tag.identifier` doit être à `STIVALE2_HEADER_TAG_FRAMEBUFFER_ID` pour signifier que ce tag est un qui donne des informations au bootloader à propos du framebuffer (taille en largeur/hauter, ...)
````
Correction proposée
"Ici, la valeur de la variable `.tag.identifier` doit être `STIVALE2_HEADER_TAG_FRAMEBUFFER_ID`. Cela signifie que ce tag donne des informations au bootloader à propos du framebuffer (taille en largeur/hauteur, ...)"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+```
+Ici la variable `.tag.identifier` doit être à `STIVALE2_HEADER_TAG_FRAMEBUFFER_ID` pour signifier que ce tag est un qui donne des informations au bootloader à propos du framebuffer (taille en largeur/hauter, ...)
+
+La variable `.tag.next` est à `0` pour l'instant car nous utilison qu'une seule entrée dans la liste.
````
Correction proposée :
"[...] nous n'utilisons qu'une seule entrée dans la liste."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+}; 
+```
+
+Maintenant nous configurons le [framebuffer](/x86_64/périphériques/framebuffer.md), nous voulons juste le mettre pour le moment en pixel et non texte, nous essayerons de remplir l'écran en bleu.
````
Correction proposée :
"Maintenant, configurons le [framebuffer](/x86_64/périphériques/framebuffer.md). Pour le moment, nous voulons le mettre en pixel et non en texte : essayons de remplir l'écran en bleu."

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+}; 
+```
+
+Ensuite vous mettez la variable `tags` du `stivale2_header` à l'adresse du tag du framebuffer soit:
````
Correction proposée :
"Ensuite, initialisez la variable `tags` du `stivale2_header` à l'adresse du tag du framebuffer soit :"

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,463 @@
+};
+```
+
+Pour finir vous devriez avoir ceci:
````
Correction proposée :
"[...] ceci :"

sleepy-monax commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -460,4 +460,16 @@ Qui donneras ce motif si tout fonctionne:
\ No newline at end of file
+Cette partie du tutoriel est terminée ! vous avez maintenant un kernel qui boot, cependant dans le prochain tutoriel nous implémenterons un driver COM, qui donnera la possibilité d'écrire des informations dans la console, ce qui est très pratique pour debugger.
+
+# Sources / Pour aller plus loins
```
Références ;)

Supercip971 replied:
fixé, merchi

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
```diff
@@ -0,0 +1,475 @@
+
+## L'Entrée
+
+Après la mise en place du header pour le bootloader nous devons programmer le point d'entrée, `kernel_start`, c'est une fonction qui ne retourne rien mais qui a un `struct stivale2_struct*` comme argument, qui donne sont des informations passé par le bootloader.
```
Correction proposée :
"[...] comme argument." Arrête ta phrase ici et commence une nouvelle. Je n'ai pas pu corriger la suite de cette ligne parce que je ne l'ai pas comprise

Supercip971 replied:
je l'ai terminée mais j'ai rajouté:  Cet argument (ici bootloader_data) représente les informations passées par le bootloader. 
je sais pas si c'est mieux ?

Procospecteur replied:
C'est parfait 👌

Procospecteur commented on src/x86_64/premiers-pas/01-hello-world.md:
````diff
@@ -0,0 +1,475 @@
+}
+```
+
+Maintenant il est conseillé de compiler et de tester le kernel, avant de continuer. Faites un `make run` il faut qu'il n'y ait pas d'erreur (que cela soit du bootloader ou de qemu).
````
Correction proposée :
"[...] Faites un `make run`, il faut qu'il y ait aucune erreur ; ni du bootloader, ni de qemu."

sleepy-monax commented:
Il faut aussi ne pas oublier de fix les erreurs de lint sauf les lignes trop longues, je crois que ça on en a rien a faire ;)

Supercip971 commented:
> Il faut aussi ne pas oublier de fix les erreurs de lint sauf les lignes trop longues, je crois que ça on en a rien a faire ;)

peut être merge et fix tout d'un coup ?

Supercip971 commented:
update ?
